test/cartesian/XAxis.spec.tsx

```diff
@@ -590,20 +643,100 @@ describe('<XAxis />', () => {
         },
       );
     });
+
+    describe('interval', () => {
+      it('should display all ticks with interval = 0', () => {
+        const { container } = render(
+          <BarChart width={300} height={300} data={data}>
+            <XAxis dataKey="x" type="number" interval={0} />
+          </BarChart>,
+        );
+        expectXAxisTicks(container, ['0', '45', '90', '135', '180']);
+      });
+
+      it('should display every second tick with interval = 1', () => {
+        const { container } = render(
+          <BarChart width={300} height={300} data={data}>
+            <XAxis dataKey="x" type="number" interval={1} />
+          </BarChart>,
+        );
+        expectXAxisTicks(container, ['0', '90', '180']);
+      });
+
+      it('should display every third tick with interval = 2', () => {
+        const { container } = render(
+          <BarChart width={300} height={300} data={data}>
+            <XAxis dataKey="x" type="number" interval={2} />
+          </BarChart>,
+        );
+        expectXAxisTicks(container, ['0', '135']);
+      });
+
+      it('should add more ticks with tickCount and then reduce them again with interval', () => {
```

Comment:
Interval can also be certain string values if we want tests for those I.e "preserveStart", etc.

Reply:
I couldn't find what difference it makes. There's an example in docs where three out of four axes are exactly the same. Can you think of good values that will make a nice test?

Reply:
`preserveEnd` always preserves the last tick and then tries to compute ticks that fit from there
`preserveStart` always preserves the first tick and then tries to compute ticks that fit from there
`preserveStartEnd` tries to do both and create ticks that fit in the middle
`equidistantPreserveStart` tries to do the same as `preserveStart` but it doesn't insert random ticks on one side but not the other

https://codesandbox.io/p/sandbox/line-chart-of-different-axis-intervals-d2kjmp?file=%2Fsrc%2FApp.tsx%3A96%2C27

Play with the width of that and you'll see the differences

![image](https://github.com/recharts/recharts/assets/25180830/caf955d6-1e91-4e95-9f28-a99b5aade0fb)